CoThings/Views/Settings/QRCodeView.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  QRCodeView.swift
+//  CoThings
+//
+//  Created by Quentin on 22/05/2020.
+//  Copyright © 2020 Rainlab. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+import AVFoundation
+import UIKit
+
+struct QRCodeView: UIViewControllerRepresentable {
+    
+    private var controller: QRCodeViewController = QRCodeViewController()
+    
+    init(onFoundCode: @escaping (String) -> Void, onCameraError: @escaping () -> Void) {
+        controller.foundQRCode = onFoundCode
+        controller.failedToOpenCamera = onCameraError
+    }
+    
+    func makeUIViewController(context: Context) -> QRCodeViewController {
+        return controller
+    }
+    
+    func updateUIViewController(_ uiViewController: QRCodeViewController, context: Context) {}
+    
+}
+
+class QRCodeViewController: UIViewController {
+    var captureSession: AVCaptureSession!
+    var previewLayer: AVCaptureVideoPreviewLayer!
+    
+    var foundQRCode: (String) -> Void = {_ in}
+    var failedToOpenCamera: () -> Void = {}
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = UIColor.black
+        captureSession = AVCaptureSession()
+
+        guard let videoCaptureDevice = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: .back) else { return }
+        let videoInput: AVCaptureDeviceInput
+
+        do {
+            videoInput = try AVCaptureDeviceInput(device: videoCaptureDevice)
+        } catch {
+            failedToOpenCamera()
+            return
+        }
+
+        if (captureSession.canAddInput(videoInput)) {
+            captureSession.addInput(videoInput)
+        } else {
+            failedToOpenCamera()
+            return
+        }
+
+        let metadataOutput = AVCaptureMetadataOutput()
+
+        if (captureSession.canAddOutput(metadataOutput)) {
+            captureSession.addOutput(metadataOutput)
+
+            metadataOutput.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+            metadataOutput.metadataObjectTypes = [.qr]
+        } else {
+            failedToOpenCamera()
+            return
+        }
+
+        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        previewLayer.frame = view.layer.bounds
+        previewLayer.videoGravity = .resizeAspectFill
+        view.layer.addSublayer(previewLayer)
+
+        captureSession.startRunning()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        if (captureSession?.isRunning == false) {
+            captureSession.startRunning()
```

Comment:
Documentation says this is a blocking call and running this method in a separate queue in the background is recommended.